decorate_test.go

```diff
@@ -34,56 +34,95 @@ func TestDecorateSuccess(t *testing.T) {
 	t.Run("simple decorate without names or groups", func(t *testing.T) {
 		t.Parallel()
 		type A struct {
-			name string
+			Name string
 		}
 
 		c := digtest.New(t)
-		c.RequireProvide(func() *A { return &A{name: "A"} })
+		c.RequireProvide(func() *A { return &A{Name: "A"} })
 
 		c.RequireInvoke(func(a *A) {
-			assert.Equal(t, "A", a.name, "expected name to not be decorated yet.")
+			assert.Equal(t, "A", a.Name, "expected name to not be decorated yet.")
 		})
 
-		c.RequireDecorate(func(a *A) *A { return &A{name: a.name + "'"} })
+		c.RequireDecorate(func(a *A) *A { return &A{Name: a.Name + "'"} })
 
 		c.RequireInvoke(func(a *A) {
-			assert.Equal(t, "A'", a.name, "expected name to equal decorated name.")
+			assert.Equal(t, "A'", a.Name, "expected name to equal decorated name.")
 		})
 	})
 
 	t.Run("simple decorate a provider from child scope", func(t *testing.T) {
 		t.Parallel()
 		type A struct {
-			name string
+			Name string
 		}
 
 		c := digtest.New(t)
 		child := c.Scope("child")
-		child.RequireProvide(func() *A { return &A{name: "A"} }, dig.Export(true))
+		child.RequireProvide(func() *A { return &A{Name: "A"} }, dig.Export(true))
 
-		child.RequireDecorate(func(a *A) *A { return &A{name: a.name + "'"} })
+		child.RequireDecorate(func(a *A) *A { return &A{Name: a.Name + "'"} })
 		c.RequireInvoke(func(a *A) {
-			assert.Equal(t, "A", a.name, "expected name to equal original name in parent scope")
+			assert.Equal(t, "A", a.Name, "expected name to equal original name in parent scope")
 		})
 
 		child.RequireInvoke(func(a *A) {
-			assert.Equal(t, "A'", a.name, "expected name to equal decorated name in child scope")
+			assert.Equal(t, "A'", a.Name, "expected name to equal decorated name in child scope")
+		})
+	})
+
+	t.Run("check parent-provided decorator doesn't need parent to invoke", func(t *testing.T) {
+		type A struct {
+			Name string
+		}
+
+		type B struct {
+			dig.In
+
+			Values []string `group:"values"`
+		}
+		type C struct {
+			dig.Out
+
+			Values []string `group:"values"`
+		}
+
+		c := digtest.New(t)
+		child := c.Scope("child")
+
+		c.RequireProvide(func() *A { return &A{Name: "A"} }, dig.Export(true))
+		c.RequireProvide(func() string { return "val1" }, dig.Export(true), dig.Group("values"))
+		c.RequireProvide(func() string { return "val2" }, dig.Export(true), dig.Group("values"))
+		c.RequireProvide(func() string { return "val3" }, dig.Export(true), dig.Group("values"))
```

Comment:
Is dig.Export needed on the root container?

Reply:
it is not. Removed